docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -661,6 +661,38 @@ This situation is very common when trying to use CDI with alternative JVM langua
 
 Quarkus however, can overcome these limitations when `quarkus.arc.transform-unproxyable-classes` is set to `true` (which is the default value).
 
+=== Container-managed Concurrency
+
+There is no standard concurrency control mechanism for CDI beans.
+Nevertheless, a bean instance can be shared and accessed concurrently from multiple threads.
+In that case it should be thread-safe.
+You can use standard Java constructs (`volatile`, `synchronized`, `ReadWriteLock`, etc.) or let the container control the concurrent access. 
+
+.Container-managed Concurrency Example
+[source,java]
+----
+import io.quarkus.arc.Lock;
+
+@Lock <1>
+@ApplicationScoped
+class SharedService {
+
+  void addAmount(BigDecimal amout) {
+    // ...changes some internal state of the bean
+  }
+  
+  @Lock(value = Lock.Type.READ, time = 1, unit = TimeUnit.SECONDS) <2> <3>
+  BigDecimal getAmount() {
+    // ...it is safe to read the value concurrently
+  }
+}
+----
+<1> `@Lock` (which maps to `@Lock(Lock.Type.WRITE)`) declared on the class instructs the container to lock the bean instance for any invocation of any business method, i.e. the client has "exclusive access" and no concurrent invocations will be allowed.
```

Comment:
Should we include a note mentioning what the default timeout is for such locks?

Reply:
AFAIK there is no timeout by default...